src/Tribe/Event_Repository.php

```diff
@@ -20,251 +26,65 @@ class Tribe__Tickets__Event_Repository extends Tribe__Repository__Decorator {
-			) ) && ! ( is_array( $value ) && 2 === count( $value ) ) ) {
-			throw Tribe__Repository__Usage_Error::because_this_comparison_operator_requires_an_value_of_type( $operator, 'filter_by_cost', 'array' );
-		}
+		// These filter methods are added by the Post_Tickets trait.
+		$this->decorated->add_schema_entry( 'cost', [ $this, 'filter_by_cost' ] );
+		$this->decorated->add_schema_entry( 'cost_currency_symbol', [ $this, 'filter_by_cost_currency_symbol' ] );
+		$this->decorated->add_schema_entry( 'has_tickets', [ $this, 'filter_by_has_tickets' ] );
+		$this->decorated->add_schema_entry( 'has_rsvp', [ $this, 'filter_by_has_rsvp' ] );
 
-		if ( in_array( $operator, array( 'IN', 'NOT IN' ) ) ) {
-			$value = (array) $value;
-		}
+		// These filter methods are added by the Post_Attendees trait.
+		$this->decorated->add_schema_entry( 'has_attendees', [ $this, 'filter_by_has_attendees' ] );
+		$this->decorated->add_schema_entry( 'attendee', [ $this, 'filter_by_attendee' ] );
+		$this->decorated->add_schema_entry( 'attendee__not_in', [ $this, 'filter_by_attendee_not_in' ] );
+		$this->decorated->add_schema_entry( 'attendee_user', [ $this, 'filter_by_attendee_user' ] );
 
-		$operator_name = Tribe__Utils__Array::get( Tribe__Repository::get_comparison_operators(), $operator, '' );
-		$prefix        = str_replace( '-', '_', 'by_cost_' . $operator_name );
-
-		global $wpdb;
-
-		// Join to the meta that relates tickets to events.
-		$this->decorated->join_clause( "JOIN {$wpdb->postmeta} {$prefix}_ticket_event 
-			ON ( 
-				{$prefix}_ticket_event.meta_value = {$wpdb->posts}.ID
-				AND {$prefix}_ticket_event.meta_key REGEXP '^_tribe_.*_for_event$' 
-			)" );
-
-		$price_regexp_frags = array(
-			// PayPal and WooCommerce tickets.
-			'_price',
-			// Easy Digital Downloads tickets.
-			'edd_price',
-		);
-		$price_regexp       = '^(' . implode( '|', $price_regexp_frags ) . ')$';
-
-		// Join to the ticket cost meta, allow for RSVP tickets too that have no price.
-		$this->decorated->join_clause( $wpdb->prepare( "LEFT JOIN {$wpdb->postmeta} {$prefix}_ticket_cost 
-			ON (
-					{$prefix}_ticket_cost.post_id = {$prefix}_ticket_event.post_id
-					AND (
-						{$prefix}_ticket_cost.meta_key REGEXP %s
-						OR {$prefix}_ticket_cost.meta_id IS NULL
-					)
-			)",
-			$price_regexp
-		) );
-
-		$prepared_value = is_array( $value )
-			? $this->decorated->prepare_interval( $value, '%d', $operator )
-			: $wpdb->prepare( '%d', $value );
-
-		// Default the cost to `0` if not set to make RSVP tickets show as "free" tickets, with a cost of 0.
-		$this->decorated->where_clause( "IFNULL( {$prefix}_ticket_cost.meta_value, 0 ) {$operator} {$prepared_value}" );
-
-		if ( null !== $symbol ) {
-			$this->filter_by_cost_currency_symbol( $symbol );
-		}
+		// This is not yet working, it needs more debugging to determine why it's not functional yet.
+		//$this->decorated->add_schema_entry( 'attendee_user__not_in', [ $this, 'filter_by_attendee_user_not_in' ] );
```

Comment:
* :no_entry_sign: No space found before comment text; expected "// $this->decorated->add_schema_entry( 'attendee_user__not_in', [ $this, 'filter_by_attendee_user_not_in' ] );" but found "//$this->decorated->add_schema_entry( 'attendee_user__not_in', [ $this, 'filter_by_attendee_user_not_in' ] );"
* :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 697127f20761b0b2886ec891b2f64307</sub>